### ISSUE TITLE
no-unnecessary-whitespace: trailing indentation before the closing backtick is incorrectly stripped

### ISSUE THREAD

**Issue by fkoepke** [sergioazoc/oxlint-tailwindcss#109]
The rule reports and removes whitespace from template literal class strings that close with a newline followed by indentation — i.e. where the closing backtick is on its own **indented** line:

```ts
    const className = `
        bg-red-500 text-white hover:bg-red-600
    `
// ^^ trailing spaces before backtick
```

Note: if the closing backtick is at **column 0** (no indentation), the trailing line is empty and the rule already handles it correctly. The bug only occurs when there are spaces before the closing backtick.

The following patterns are already handled correctly (both pass):

```ts
// Opening + closing newline, backtick at column 0 — VALID, no report ✓
{ code: 'const className = `\n  bg-red-500 text-white\n  hover:bg-red-600 focus:ring-2\n`', filename: 'test.tsx' },

// Opening newline only — VALID, no report ✓
{ code: 'const className = `\n  bg-red-500 text-white hover:bg-red-600`', filename: 'test.tsx' },
```

The following pattern should also be valid:

```ts
// Closing newline with indentation before backtick — should be VALID, currently reports ✗
{ code: 'const className = `\n    bg-red-500 text-white hover:bg-red-600\n    `', filename: 'test.tsx' },
```

**Comment by sergioazoc** [comment]
Thanks for the precise report, @fkoepke. You pinned the exact case: the trailing-edge trim was
eating one space of the closing backtick's indentation, and a whitespace-only last line is
intentional block formatting, not a stray space.

Fixed in #113, shipping in v1.8.0. It also turned out to be a prerequisite for the block-convention
autofix in #111 (a nested block ends in `\n` + indent), so this one pulled double duty. Appreciate
the detailed repro cases.
